docs/landscape.md

```diff
@@ -0,0 +1,54 @@
+# Prow Landscape
+In this page, the landscape of the Prow setup is documented. It outlines the different systems and components involved and the interconnection between them.
+
+The "Prow cluster" is a Kubernetes instance managed by Google Kubernetes Engine as part of the project, called "kyma-project". In that cluster, the Prow components are installed.
+
+![Landscape overview](assets/landscape.svg)
+
+# Provisioning and Secrets
+The cluster gets provisioned by a script executed manually by an _admin_. The script will read the full configuration from the test-infra repository so that the cluster can be re-created at any time. Therefore, all the configuration changes should be applied to the repository instead of the runtime.
```

Comment:
Perhaps it could be good to mention which provisioning script is responsible for that?